### DISCUSSION TITLE
Ctrl+C kills plink, but I want to kill candump. Terminal -> plink -> candump

### DISCUSSION THREAD

**Discussion by vslynko** [microsoft/terminal#14126] [Q&A] [answered]
Hi everyone,
I have a question regarding passing the signals down the chain. I have a setup where I run a plink session inside Windows Terminal.
That plink session connected to the debug interface of some hardware with yocto linux. I run a candump tool inside the debug session to sniff on the canbus communication. But if I click Ctrl+C, the plink session terminates and terminal tab gets closed, the candump session, however, keeps running on the remote. I can run plink again and connect to the same session.

How can I pass sigkill to candump instead of killing teh whole debugging.

If I use Putty directly, Ctrl+C works as expected.

**Comment by zadjii-msft** [comment, marked as answer]
I was under the impression that plink wasn't actually designed for interactive use: https://github.com/microsoft/terminal/issues/10943#issuecomment-899567471 has some discussion. I coulda swore there was a SO post that had more details, but I can't find it. I thought is was more for scripts than for interactive use from the console/terminal. It might not be designed to actually handle a Ctrl+C from the console.

**Comment by touchmii** [comment]
> Hi everyone, I have a question regarding passing the signals down the chain. I have a setup where I run a plink session inside Windows Terminal. That plink session connected to the debug interface of some hardware with yocto linux. I run a candump tool inside the debug session to sniff on the canbus communication. But if I click Ctrl+C, the plink session terminates and terminal tab gets closed, the candump session, however, keeps running on the remote. I can run plink again and connect to the same session.
> 
> How can I pass sigkill to candump instead of killing teh whole debugging.
> 
> If I use Putty directly, Ctrl+C works as expected.

you can try ss, https://github.com/fasteddy516/SimplySerial.

**Reply by nicowilliams** [reply]
The issue must be that `plink` doesn't know how to put the Windows Terminal console into raw mode (and later how to restore it to cooked mode).

It should be easy to contribute a fix to the PuTTY project.  https://claude.ai/share/5983f69d-9b5b-4098-a2da-833c5f1eb093
